gridstatus/spp.py

```diff
@@ -707,6 +707,56 @@ def _process_operating_reserves(self, df):
 
         return df
 
+    @support_date_range("DAY_START")
+    def get_day_ahead_marginal_clearing_prices(self, date, end=None, verbose=False):
+        """Provides Marginal Clearing Price information by Reserve Zone for each
+        Day-Ahead Market solution for each Operating Day.
+        Posting is updated each day after the DA Market results are posted.
+        Available at https://portal.spp.org/pages/da-mcp#
+
+        Args:
+            date: date to get data for
+            end: end date
+            verbose: print url
+
+        Returns:
+            pd.DataFrame: Day Ahead Marginal Clearing Prices
+        """
+        if date == "latest":
+            raise ValueError(
+                "Latest not supported for Day Ahead Marginal Clearing Prices",
+            )
+
+        url = f"{FILE_BROWSER_DOWNLOAD_URL}/da-mcp?path=/{date.strftime('%Y')}/{date.strftime('%m')}/DA-MCP-{date.strftime('%Y%m%d')}0100.csv"  # noqa
+
+        msg = f"Downloading {url}"
+        log(msg, verbose)
+        df = pd.read_csv(url)
+
+        return self._process_day_ahead_marginal_clearing_prices(df)
+
+    def _process_day_ahead_marginal_clearing_prices(self, df):
+        df = self._handle_market_end_to_interval(
+            df,
+            column="GMTIntervalEnd",
+            interval_duration=pd.Timedelta(hours=1),
+        )
+
+        column_mapping = {
+            "RegUP": "Reg_Up",
+            "RegDN": "Reg_Dn",
+            "RampUP": "Ramp_Up",
+            "RampDN": "Ramp_Dn",
+            "UncUP": "Unc_Up",
+            "Spin": "Spin",
+            "Supp": "Supp",
+        }
+
+        return df.rename(columns=column_mapping)[
+            ["Time", "Interval Start", "Interval End", "Reserve Zone"]
+            + list(column_mapping.values())
```

Comment:
actually, i think this logic is going to fail when you try to load in historical annual files that don't have all these columns

Reply:
the historical annual files are missing some columns, presumably because those operating reserve types didnt exist back then

Reply:
What is the right way to handle this situation? Add in the columns with all nulls or simply not select them if they don't exist? 

A. 
```python
cols_to_keep = [...]
for col in cols_to_keep:
    if col not in df:
         df[col] = np.nan

df[cols_to_keep]
```

B.
```python
cols_to_keep = [...]
df[[col for col in cols_to_keep if col in df]]
```

Reply:
i would do B to best reflect the raw data

when we load in our database we can add the columns that are missing with nulls.

Reply:
Yeah, it looks like `common.ensure_columns`  adds the columns if they are missing.